evm/src/cpu/kernel/asm/core/terminate.asm

```diff
@@ -205,3 +223,41 @@ global terminate_common:
 
     // stack: parent_pc, success, leftover_gas
     JUMP
+
+
+
+
+// Returns 1 if the address is present in SEGMENT_CREATED_CONTRACTS, meaning that it has been
+// created during this transaction. Returns 0 otherwise.
+// Pre stack: addr
+// Post stack: is_just_created
+%macro contract_just_created
+    // stack: addr
+    %mload_global_metadata(@GLOBAL_METADATA_CREATED_CONTRACTS_LEN)
+    // stack: nb_created_contracts, addr
+    PUSH 0
+%%maybe_insert_selfdestruct_list_loop:
```

Comment:
nit: This name doesn't really make sense since this macro doesn't insert in the selfdestruct list.